sdk/core/azure-core/src/main/java/com/azure/android/core/util/logging/ClientLogger.java

```diff
@@ -0,0 +1,289 @@
+                }
+            }
+        }
+
+        switch (logLevel) {
+            case VERBOSE_LEVEL:
+                logger.debug(format, args);
+                break;
+            case INFORMATIONAL_LEVEL:
+                logger.info(format, args);
+                break;
+            case WARNING_LEVEL:
+                if (!isNullOrEmpty(throwableMessage)) {
+                    format += System.lineSeparator() + throwableMessage;
+                }
+                logger.warn(format, args);
+                break;
+            case ERROR_LEVEL:
+                if (!isNullOrEmpty(throwableMessage)) {
+                    format += System.lineSeparator() + throwableMessage;
+                }
+                logger.error(format, args);
+                break;
+            default:
+                // Don't do anything, this state shouldn't be possible.
+                break;
+        }
+    }
+
+    /*
+     * Determines if the environment and logger support logging at the given log level.
+     *
+     * @param logLevel Logging level for the log message.
+     * @return Flag indicating if the environment and logger are configured to support logging at the given log level.
+     */
+    private boolean canLogAtLevel(int logLevel) {
+        // Determine if the logger configuration supports logging at the level.
+        switch (logLevel) {
+            case VERBOSE_LEVEL:
+                return logger.isDebugEnabled();
+            case INFORMATIONAL_LEVEL:
+                return logger.isInfoEnabled();
+            case WARNING_LEVEL:
+                return logger.isWarnEnabled();
+            case ERROR_LEVEL:
+                return logger.isErrorEnabled();
+            default:
+                return false;
+        }
+    }
+
+    /*
+     * Determines if the arguments contains a throwable that would be logged, SLF4J logs a throwable if it is the last
+     * element in the argument list.
+     *
+     * @param args The arguments passed to format the log message.
+     * @return True if the last element is a throwable, false otherwise.
+     */
+    private boolean doesArgsHaveThrowable(Object... args) {
+        if (args.length == 0) {
+            return false;
+        }
+
+        return args[args.length - 1] instanceof Throwable;
+    }
+
+    /*
+     * Removes the last element from the arguments as it is a throwable.
+     *
+     * @param args The arguments passed to format the log message.
+     * @return The arguments with the last element removed.
+     */
+    private Object[] removeThrowable(Object... args) {
+        return Arrays.copyOf(args, args.length - 1);
+    }
+
+    /**
+     * Checks if the character sequence is null or empty.
+     * @param charSequence Character sequence being checked for nullness or emptiness.
+     * @return True if the character sequence is null or empty, false otherwise.
+     */
+    private static boolean isNullOrEmpty(CharSequence charSequence) {
```

Comment:
I'm including a CoreUtils interface with static methods in PR #139 that includes this method, I think we can remove it from here.